op-dispute-mon/mon/claims.go

```diff
@@ -0,0 +1,86 @@
+package mon
+
+import (
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-dispute-mon/metrics"
+	"github.com/ethereum-optimism/optimism/op-dispute-mon/mon/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type RClock interface {
+	Now() time.Time
+}
+
+type ClaimMetrics interface {
+	RecordClaims(status metrics.ClaimStatus, count int)
+}
+
+type ClaimMonitor struct {
+	logger  log.Logger
+	clock   RClock
+	metrics ClaimMetrics
+}
+
+func NewClaimMonitor(logger log.Logger, clock RClock, metrics ClaimMetrics) *ClaimMonitor {
+	return &ClaimMonitor{logger, clock, metrics}
+}
+
+func (c *ClaimMonitor) CheckClaims(games []*types.EnrichedGameData) {
+	claimStatus := make(map[metrics.ClaimStatus]int)
```

Comment:
nit: The set of claim statuses is small enough to fit in a struct. Could we use a struct rather than a map here so it's more legible.

Reply:
Definitely!

Reply:
Actually attempted this and it became very ugly since we then have to explicitly record metrics for each struct field and can't perform an idiomatic loop across the `metrics.ClaimStatus` keys in the map.

Would prefer to do it this way so there's not too much bloat, having both a struct and an enum list doesn't seems very clean and we previously moved away from structs to enums in other dispute mon metrics.